### PR TITLE
chore: liq0/liq1 table driven test framework

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #3156 

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

*(E.g.: This pull request improves documation of area A by adding ....*

## Brief Changelog

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/math_test.go [hunk 1]
```diff
@@ -6,17 +6,87 @@ import (
...
+		name              string
+		currentSqrtP      sdk.Dec
+		sqrtPHigh         sdk.Dec
+		amount0Desired    sdk.Int
+		expectedLiquidity string
+	}{
+		{
+			"happy path",
+			sdk.NewDecWithPrec(70710678, 6),
+			sdk.NewDecWithPrec(74161984, 6),
```
Comment (mattverse) [inline x/concentrated-liquidity/math_test.go:78]: ditto

File: x/concentrated-liquidity/math_test.go [hunk 2]
```diff
@@ -6,17 +6,87 @@ import (
...
-	amount0Desired := sdk.NewInt(1)
-	amount1Desired := sdk.NewInt(5000)
+	for _, tc := range testCases {
+		tc := tc
 
-	s.SetupTest()
+		suite.Run(tc.name, func() {
+			liquidity := cl.GetLiquidityFromAmounts(tc.currentSqrtP, tc.sqrtPLow, tc.sqrtPHigh, tc.amount0Desired, tc.amount1Desired)
+			suite.Require().Equal(tc.expectedLiquidity, liquidity.String())
+		})
```
Comment (mattverse) [inline x/concentrated-liquidity/math_test.go:36]: can we add a check to indicate that this is same as result from `Liquidity1`, which is the min value?

File: x/concentrated-liquidity/math_test.go [hunk 3]
```diff
@@ -6,17 +6,87 @@ import (
...
+		expectedLiquidity string
+	}{
+		{
+			"happy path",
+			sdk.NewDecWithPrec(70710678, 6),
+			sdk.NewDecWithPrec(74161984, 6),
+			sdk.NewDecWithPrec(67082039, 6),
+			sdk.NewInt(1),
+			sdk.NewInt(5000),
+			"1377.927096082029653542",
```
Comment (mattverse) [inline x/concentrated-liquidity/math_test.go:26]: are these test case results reverse engineered as well?

**Review by mattverse — COMMENTED** [review]
Left some initial comments, please take a look! 🌮

File: x/concentrated-liquidity/math_test.go (hunk 3 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/math_test.go:26]: yes!

**Review by mattverse — APPROVED** [review]

**Pull request merged**